### ISSUE TITLE
NameError: uninitialized constant BeValidAsset::RSpec

### ISSUE THREAD

**Issue by HaleTom** [unboxed/be_valid_asset#18]
This could be a noob question:

Is there a way to use this gem without using Rspec, but from rake test or rails test instead?

In `test/test_helper.rb` I have
```
require 'be_valid_asset.rb'                                                               
include BeValidAsset                                                                      
BeValidAsset::Configuration.display_invalid_content = true                                
BeValidAsset::Configuration.enable_caching = true                                         
BeValidAsset::Configuration.cache_path = Rails.root.join('tmp', 'be_valid_asset_cache')   
```

```
ravi@boxy:~/prj/sample_app(Ch8-basic-login *)$ bundle exec rake test                                                                                                                 
rake aborted!
NameError: uninitialized constant BeValidAsset::RSpec
/home/ravi/.gem/ruby/2.3.1/gems/be_valid_asset-1.3.0/lib/be_valid_asset/be_valid_base.rb:4:in `<module:BeValidAsset>'
/home/ravi/.gem/ruby/2.3.1/gems/be_valid_asset-1.3.0/lib/be_valid_asset/be_valid_base.rb:2:in `<top (required)>'
/home/ravi/.gem/ruby/2.3.1/gems/be_valid_asset-1.3.0/lib/be_valid_asset.rb:3:in `<top (required)>'
/home/ravi/prj/sample_app/test/test_helper.rb:13:in `<top (required)>'
/home/ravi/prj/sample_app/test/controllers/sessions_controller_test.rb:1:in `<top (required)>'
/home/ravi/.gem/ruby/2.3.1/gems/railties-5.0.0/lib/rails/test_unit/test_requirer.rb:11:in `block in require_files'
/home/ravi/.gem/ruby/2.3.1/gems/railties-5.0.0/lib/rails/test_unit/test_requirer.rb:10:in `each'
/home/ravi/.gem/ruby/2.3.1/gems/railties-5.0.0/lib/rails/test_unit/test_requirer.rb:10:in `require_files'
/home/ravi/.gem/ruby/2.3.1/gems/railties-5.0.0/lib/rails/test_unit/minitest_plugin.rb:76:in `plugin_rails_init'
/home/ravi/.gem/ruby/2.3.1/gems/minitest-5.9.0/lib/minitest.rb:80:in `block in init_plugins'
/home/ravi/.gem/ruby/2.3.1/gems/minitest-5.9.0/lib/minitest.rb:78:in `each'
/home/ravi/.gem/ruby/2.3.1/gems/minitest-5.9.0/lib/minitest.rb:78:in `init_plugins'
/home/ravi/.gem/ruby/2.3.1/gems/minitest-5.9.0/lib/minitest.rb:129:in `run'
/home/ravi/.gem/ruby/2.3.1/gems/railties-5.0.0/lib/rails/test_unit/minitest_plugin.rb:64:in `rake_run'
/home/ravi/.gem/ruby/2.3.1/gems/railties-5.0.0/lib/rails/test_unit/testing.rake:15:in `block in <top (required)>'
/home/ravi/.gem/ruby/2.3.1/gems/rake-11.2.2/exe/rake:27:in `<top (required)>'
/home/ravi/.gem/ruby/2.3.1/gems/bundler-1.12.5/lib/bundler/cli/exec.rb:63:in `load'
/home/ravi/.gem/ruby/2.3.1/gems/bundler-1.12.5/lib/bundler/cli/exec.rb:63:in `kernel_load'
/home/ravi/.gem/ruby/2.3.1/gems/bundler-1.12.5/lib/bundler/cli/exec.rb:24:in `run'
/home/ravi/.gem/ruby/2.3.1/gems/bundler-1.12.5/lib/bundler/cli.rb:304:in `exec'
/home/ravi/.gem/ruby/2.3.1/gems/bundler-1.12.5/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
/home/ravi/.gem/ruby/2.3.1/gems/bundler-1.12.5/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
/home/ravi/.gem/ruby/2.3.1/gems/bundler-1.12.5/lib/bundler/vendor/thor/lib/thor.rb:359:in `dispatch'
/home/ravi/.gem/ruby/2.3.1/gems/bundler-1.12.5/lib/bundler/vendor/thor/lib/thor/base.rb:440:in `start'
/home/ravi/.gem/ruby/2.3.1/gems/bundler-1.12.5/lib/bundler/cli.rb:11:in `start'
/home/ravi/.gem/ruby/2.3.1/gems/bundler-1.12.5/exe/bundle:27:in `block in <top (required)>'
/home/ravi/.gem/ruby/2.3.1/gems/bundler-1.12.5/lib/bundler/friendly_errors.rb:98:in `with_friendly_errors'
/home/ravi/.gem/ruby/2.3.1/gems/bundler-1.12.5/exe/bundle:19:in `<top (required)>'
/home/ravi/.gem/ruby/2.3.1/bin/bundle:23:in `load'
/home/ravi/.gem/ruby/2.3.1/bin/bundle:23:in `<main>'
Tasks: TOP => test
(See full trace by running task with --trace)
ravi@boxy:~/prj/sample_app(Ch8-basic-login *)[1]$ 
```

**Comment by HaleTom** [comment]
I found the one-line fix in:
https://github.com/unboxed/be_valid_asset/pull/17/files

@h-lame Could you or one of the team pull this?

**Comment by HaleTom** [comment]
Sorry, I bounced accidentally on the close button.

Leaving this open pending the pull.

**Comment by h-lame** [comment]
Hi @HaleTom 

Just merged in that PR and released v1.3.1, thanks for prompting us.  

However, this won't let you use the gem without rspec; the `be_valid_xxx` matchers that this gem provides are based on rspec and you'll still need the rspec gem installed to use them.  I don't really know if they'll work in a minitest / test::unit test suite as they're competeing frameworks that do things differently to rspec.

If it doesn't work it might be possible to separate the validation code from the rspec integration code so it's easier to use the underlying validation service without requiring rspec.  This is not something we'd be likely to investigate ourselves as we don't use minitest or test::unit on any projects, but we'd be happy to review a PR if you wanted to take a look yourself.

Cheers,

Murray
